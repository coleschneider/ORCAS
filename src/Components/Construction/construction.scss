@import '../../stylesheets/layout/colors.scss';
  .shade {
    position: absolute;
    background: rgba(0, 0, 0, 0.8);
    top: 0;
    left: 0;
    z-index: 3;
    right: 0;
    bottom: 0;
  }
  
  .modal {
    z-index: 4;
    position: absolute;
    // width: 80%;
    right: 0;
    left: 0;
    background: white;
    border-radius: 10px;
    top: 10%;
    height: 50vh;
    
  }
  .modal_content-wrapper {
    display: flex; 
    padding: 30px;
    flex-direction: column;
    align-items: center;
  }
  .modal_text-wrapper {
    font-size: 2.1rem;
    color: $dark-sky;
    font-weight: 800;
  }
  .modal_content-container {
    flex-direction: column;
    display: flex;
  }
  .modal_button-group {
   align-items: stretch;
   display: flex;
   flex-direction: column;
  }

  .modal_button {
     display: flex;
     align-items: center;
     margin: 10px 10px;
     justify-content: center;
     border: 1px solid $bg-grey;
     font-weight: 600;
     cursor: pointer;
  }
  
.button-1 {
    flex: 1;
  }
  .button-2 {
    flex: 1;
  }


  @media only screen and (min-width: 800px){
    .modal {
      right: 10%;
      left: 10%;
      // width: 50%;
      height: 300px;
      bottom: 50%;
    }
    .modal_button-group {
      align-items: stretch;
      display: flex;
      flex-direction: row;
     }
  }