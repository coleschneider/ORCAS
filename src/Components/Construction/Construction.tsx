import './construction.scss';
import React from 'react';
import posed, { PoseGroup } from 'react-pose';
import { Link } from 'react-scroll';

const Modal = posed.div({
  enter: {
    default: { duration: 300 },
    delay: 300,
    opacity: 1,
    transition: {
      y: { damping: 15, type: 'spring', stiffness: 1000 },
    },
    y: 0,
  },
  exit: {
    opacity: 0,
    transition: { duration: 150 },
    y: 100,
  },
});

const Shade = posed.div({
  enter: { opacity: 1 },
  exit: { opacity: 0 },
});

export default class Example extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isConstruction: false,
    };
  }

  componentDidMount() {
    this.toggleModal();
  }

  toggleModal = () => {
    this.setState(
      {
        isConstruction: !this.state.isConstruction,
      },
      () => {
        if (this.state.isConstruction) return (document.body.style.overflow = 'hidden');
        document.body.style.overflow = '';
      },
    );
  };

  render() {
    const { isConstruction } = this.state;
    return (
      <PoseGroup>
        {isConstruction && [
          <Shade key="shade" className="shade" />,
          <Modal key="modal" className="modal">
            <div className="modal_content-wrapper">
              <div className="modal_content-container">
                <div className="modal_text-wrapper">ORCASENIORS.org is currently under construction.</div>
                <div className="modal_text-sub">Please leave your email if you wish to be contacted.</div>
                <div className="modal_button-group">
                  <Link
                    className="modal_button button-1"
                    to="contact"
                    onClick={() => this.toggleModal()}
                    spy={true}
                    smooth={true}
                    offset={-70}
                    duration={500}
                  >
                    Sign up for newsletter
                  </Link>
                  <a className="modal_button button-1" onClick={() => this.toggleModal()}>
                    Continue to site
                  </a>
                </div>
              </div>
            </div>
          </Modal>,
        ]}
      </PoseGroup>
    );
  }
}
