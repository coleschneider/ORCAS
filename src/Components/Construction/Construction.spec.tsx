import * as React from 'react';
import Construction from './Construction';

describe('Construction Modal', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Construction />);
    expect(wrapper).toMatchSnapshot();
  });
});
