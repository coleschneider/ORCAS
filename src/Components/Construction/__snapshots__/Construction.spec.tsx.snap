// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Construction Modal should match the snapshot 1`] = `
<PoseGroup
  flipMove={true}
>
  <ForwardRef
    className="shade"
    key="shade"
  />
  <ForwardRef
    className="modal"
    key="modal"
  >
    <div
      className="modal_content-wrapper"
    >
      <div
        className="modal_content-container"
      >
        <div
          className="modal_text-wrapper"
        >
          ORCASENIORS.org is currently under construction.
        </div>
        <div
          className="modal_text-sub"
        >
          Please leave your email if you wish to be contacted.
        </div>
        <div
          className="modal_button-group"
        >
          <Link
            className="modal_button button-1"
            duration={500}
            offset={-70}
            onClick={[Function]}
            smooth={true}
            spy={true}
            to="contact"
          >
            Sign up for newsletter
          </Link>
          <a
            className="modal_button button-1"
            onClick={[Function]}
          >
            Continue to site
          </a>
        </div>
      </div>
    </div>
  </ForwardRef>
</PoseGroup>
`;
